src/library/scala/io/Codec.scala

```diff
@@ -77,17 +64,20 @@ trait LowPriorityCodecImplicits {
 }
 
 object Codec extends LowPriorityCodecImplicits {
-  final val ISO8859: Codec = new Codec(Charset forName "ISO-8859-1")
-  final val UTF8: Codec    = new Codec(Charset forName "UTF-8")
+  @deprecated("this method is misnamed and will be removed, use `Codec(StandardCharsets.ISO_8859_1) instead`", "2.12.0")
+  final val ISO8859: Codec = new Codec(StandardCharsets.ISO_8859_1)
+  final val UTF8: Codec    = new Codec(StandardCharsets.UTF_8)
 
-  /** Optimistically these two possible defaults will be the same thing.
+  /* Optimistically these two possible defaults will be the same thing.
    *  In practice this is not necessarily true, and in fact Sun classifies
    *  the fact that you can influence anything at all via -Dfile.encoding
    *  as an accident, with any anomalies considered "not a bug".
    */
-  def defaultCharsetCodec = apply(Charset.defaultCharset)
+  /** Always returns a UTF8 codec. */
+  def defaultCharsetCodec = UTF8
+  @deprecated("use `Codec(scala.util.Properties.encodingString)` to create a codec influenced by the `-Dfile.encoding` JVM flag", "2.12.0")
```

Comment:
I wonder if all file.encoding usage should be deprecated. Also on util.Properties. It's a naming conflict with jvm internals and a bug to use it.